docs/design/K2_BASIC_CONCEPTS.md

```diff
@@ -0,0 +1,175 @@
+struct KeyRangeVersion {
+    // the starting key for the partition
+    String startKey;
+    // the ending key for the partition
+    String endKey;
+    // the partition version
+    PVID pvid;
+};
+
+// Partition in a K2 Collection. By default, the key-range type is String (for range-based partitioning)
+// but it can also be an integral type for hash-based partitioning
+struct Partition {
+    // the key range version
+    KeyRangeVersion keyRangeV;
+    // the endpoints for the node which is currently assigned to this partition(version)
+    std::set<String> endpoints;
+    // the current assignment state of the partition
+    AssignmentState astate = AssignmentState::NotAssigned;
+};
+
+struct PartitionMap {
+    uint64_t version =0;
+    std::vector<Partition> partitions;
+};
+
+struct CollectionCapacity {
+    uint64_t dataCapacityMegaBytes = 0;
+    uint64_t readIOPs = 0;
+    uint64_t writeIOPs = 0;
+};
+
+K2_DEF_ENUM(HashScheme,
+    Range,
+    HashCRC32C
+);
+
+K2_DEF_ENUM(StorageDriver,
+            K23SI);
+
+struct CollectionMetadata {
+    String name;
+    HashScheme hashScheme;
+    StorageDriver storageDriver;
+    CollectionCapacity capacity;
+    Duration retentionPeriod{0};
+    Duration heartbeatDeadline{0}; // set by the CPO
+    // This is used by the CPO only. If deleted is true the CPO will not return the collection
+    // for getCollection RPCs, but the user can try to offload it again.
+    bool deleted{false};
+};
+
+struct Collection {
+    PartitionMap partitionMap;
+    std::unordered_map<String, String> userMetadata;
+    CollectionMetadata metadata;
+};
+
+struct Schema {
+    String name;
+    uint32_t version = 0;
+    std::vector<SchemaField> fields;
+
+    // All key fields must come before all value fields (by index), so that a key can be
+    // constructed for a read request without knowing the schema version
+    std::vector<uint32_t> partitionKeyFields;
+    std::vector<uint32_t> rangeKeyFields;
+};
+```
+
+
+## Quick overview
+This is an overview of how data is organized in K2. Details on each part are following
+
+![Data Deployment](../images/Data_Deployment.png)
+
+## Cluster, nodes, nodepool
+A recap from the K2PROCESS.md document:
+A cluster is a set of machines used to deploy data which is managed as a single unit. Wd do not have a notion of cluster in code - it is just how we may refer in documentation to a set of machines which are used to deploy a `Collection`(see below). The data lives in partitions, each deployed to a single CPU Core with some particular fixed memory available to it (4-12GB). We call this a `node` and is what messages in the system are send to. We start these on a given machine as individual threads in a process which we call the `nodepool`. Currently we deploy nodepools one per CPU socket, so a 2-socket machine (each socket with say 20cores) would get 2 nodepools, each running 20 threads for a total of 40 nodes available for deployment on this machine.
```

Comment:
Typos: "Wd"->"We" in second sentence. "messages in the system are send to" -> "messages in the system are sent to".

Reply:
fixed